apps/hash-integration-worker/src/linear-activities.ts

```diff
@@ -246,32 +254,46 @@ const createOrUpdateHashEntity = async (params: {
           },
         }),
       ),
-      ...addedOutgoingLinks.map(({ linkEntityTypeId, destinationEntityId }) =>
-        HashLinkEntity.create(graphApiClient, params.authentication, {
-          entityTypeIds: [linkEntityTypeId],
-          linkData: {
-            leftEntityId: existingEntity.metadata.recordId.entityId,
-            rightEntityId: destinationEntityId,
-          },
-          properties: { value: {} },
-          provenance,
-          webId: params.webId,
-          draft: false,
-          relationships: [
+      ...addedOutgoingLinks.map(
+        async ({ linkEntityTypeId, destinationEntityId }) => {
+          const linkEntity = await HashLinkEntity.create(
+            graphApiClient,
+            params.authentication,
             {
-              relation: "setting",
-              subject: { kind: "setting", subjectId: "administratorFromWeb" },
+              entityTypeIds: [linkEntityTypeId],
+              linkData: {
+                leftEntityId: existingEntity.metadata.recordId.entityId,
+                rightEntityId: destinationEntityId,
+              },
+              properties: { value: {} },
+              provenance,
+              webId: params.webId,
+              draft: false,
+              relationships: createLinearHashEntityAuthRelationships({
+                machineActorId: params.authentication.actorId,
+              }),
             },
-            {
-              relation: "setting",
-              subject: { kind: "setting", subjectId: "updateFromWeb" },
+          );
+
+          const linkEntityUuid = extractEntityUuidFromEntityId(
+            linkEntity.entityId,
+          );
+
+          await createPolicy(graphApiClient, params.authentication, {
```

Comment:
see https://github.com/hashintel/hash/pull/7405
```suggestion
          // TODO: allow creating policies alongside entity creation
          //   see https://linear.app/hash/issue/H-4622/allow-creating-policies-alongside-entity-creation
          await createPolicy(graphApiClient, params.authentication, {
```